apikana-maven-plugin/src/main/java/org/swisspush/apikana/AbstractApikanaMojo.java

```diff
@@ -172,23 +222,121 @@ protected boolean isPom() {
+            customConfig.put("mavenGroupId", mavenProject.getGroupId());
+            final Map<String, Object> avro = (Map) customConfig.merge("avro", new LinkedHashMap<>(), (oldVal, newVal) -> oldVal);
+            avro.put("enumAsString", true);
+            customConfig.put("avro", avro);
             pack.put("customConfig", customConfig);
-            devDependencies.put("apikana", version);
-            devDependencies.put("apikana-defaults", apikanaDefaultsVersion);
         });
     }
 
+    private String getTitle(Map<String, Object> apiSpec) {
+        Map<String, Object> info = (Map<String, Object>) apiSpec.get("info");
+        return (String) info.get("title");
+    }
+
+    private String getAuthor(Map<String, Object> apiSpec) {
+        Map<String, Object> info = (Map<String, Object>) apiSpec.get("info");
+        Map<String, Object> contact = (Map<String, Object>) info.get("contact");
+
+        StringBuilder authorBuilder = new StringBuilder();
+
+        if (contact.get("name") != null) {
+            authorBuilder.append(contact.get("name"));
+        }
+
+        if (contact.get("email") != null) {
+            if (authorBuilder.length() > 0) {
+                authorBuilder.append(" <")
+                        .append(contact.get("email"))
+                        .append(">");
+            } else {
+                authorBuilder.append(contact.get("email"));
+            }
+        }
+
+        if (contact.get("url") != null) {
+            if (authorBuilder.length() > 0) {
+                authorBuilder.append(" (")
+                        .append(contact.get("url"))
+                        .append(")");
+            } else {
+                authorBuilder.append(contact.get("url"));
+            }
+        }
+
+        return authorBuilder.toString();
+    }
+
+    private Map<String, Object> parseApiSpec() {
+        try {
+            getLog().info(String.format("basedir: %s", mavenProject.getBasedir().getAbsolutePath()));
+            getLog().info(String.format("api: %s", api));
+            File apiSpecFile = new File(mavenProject.getBasedir(), api);
+            getLog().info(String.format("apiSpecFile: %s", apiSpecFile.getAbsolutePath()));
+            if (api.endsWith(".json")) {
+                return parseJsonApiSpec(apiSpecFile);
+            } else {
+                return parseYamlApiSpec(apiSpecFile);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Map<String, Object> parseYamlApiSpec(File yamlSpecFile) throws IOException {
+        Yaml yaml = new Yaml();
+        try (InputStream inputStream = Files.newInputStream(yamlSpecFile.toPath())) {
+            return yaml.load(inputStream);
+        }
+    }
+
+    private Map<String, Object> parseJsonApiSpec(File jsonSpecFile) throws IOException {
+        final ObjectMapper mapper = new ObjectMapper();
+        return mapper.readValue(jsonSpecFile, Map.class);
```

Comment:
> final ObjectMapper mapper = new ObjectMapper();

(*minor*) Looks as this instances should be a field instead? Or why do we need yet another instance for every call?

Reply:
I created a field and used it throughout the class.